### PR TITLE
refactor(server): auth dto

### PR DESCRIPTION
**Pull request by jrasm91**
Restructure (simplify/standardize) the `AuthDto`, which is used everywhere.

### CODE REVIEW

File: server/src/domain/auth/auth.dto.ts [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
-  isShowMetadata?: boolean;
-  accessTokenId?: string;
-  externalPath?: string | null;
+export class AuthDto {
+  user!: UserEntity;
+
+  apiKey?: APIKeyEntity;
+  sharedLink?: SharedLinkEntity;
+  userToken?: UserTokenEntity;
 }
```
Comment (jrasm91) [inline server/src/domain/auth/auth.dto.ts:12]: This is the bulk of the change, the rest of it is just renaming/updating accessors.

File: server/src/immich/app.guard.ts [hunk 2]
```diff
@@ -93,12 +93,12 @@ export class AppGuard implements CanActivate {
     const req = context.switchToHttp().getRequest<AuthRequest>();
 
     const authDto = await this.authService.validate(req.headers, req.query as Record<string, string>);
-    if (authDto.isPublicUser && !isSharedRoute) {
+    if (authDto.sharedLink && !isSharedRoute) {
```
Comment (alextran1502) [inline server/src/immich/app.guard.ts:96]: What is the alternative mechanism to determine if a user is a public user?

Comment (jrasm91) [reply server/src/immich/app.guard.ts:96]: auth via a shared link implies the user is a "public user".

**Review by alextran1502 — APPROVED** [review]

**Review by danieldietzler — COMMENTED** [review]
1k loc LOL

**Pull request merged**